src/w3iProxy/notifyProviders/internalNotifyProvider.ts

```diff
@@ -187,6 +204,23 @@ export default class InternalNotifyProvider implements W3iNotifyProvider {
       throw new Error(this.formatClientRelatedError('update'))
     }
 
+    const account = this.notifyClient.subscriptions.keys.includes(params.topic)
+      ? this.notifyClient.subscriptions.get(params.topic).account
+      : ''
+
+    if (
+      !this.notifyClient.isRegistered({ account, domain: window.location.hostname, allApps: true })
+    ) {
```

Comment:
I think we shouldn't do calls inside the conditions for better readability.

```suggestion
     const isRegistered = this.notifyClient.isRegistered({
      account,
      domain: window.location.hostname,
      allApps: true
    })

    if (!isRegistered) {
```